openeo-geotrellis/src/main/scala/org/openeo/geotrellis/vector/VectorCubeMethods.scala

```diff
@@ -41,56 +57,128 @@ object VectorCubeMethods {
+    val band: RDD[(SpatialKey, Tile)] = fromKeyedFeature[Geometry](keyedFeatures, cellType, targetLayout, options)
+    band
   }
 
   /**
-   * Convert a vector datacube to a raster datacube.
+   * Convert a vector datacube to a raster datacube without a time dimension.
    *
    * @param path: Path to the geojson file.
    * @param targetDatacube: Target datacube to extract the crs and resolution from.
    * @return A raster datacube.
    */
-  def vectorToRasterGeneric[K: SpatialComponent: ClassTag](path: String, targetDatacube: MultibandTileLayerRDD[K]): MultibandTileLayerRDD[SpatialKey] = {
-    val sc = SparkContext.getOrCreate()
-    val target_layout: LayoutDefinition = targetDatacube.metadata.layout
-    val target_crs: CRS = targetDatacube.metadata.crs
-    val features: Seq[Feature[Geometry, Double]] = extractFeatures(path, target_crs, target_layout)
-    val featuresRDD: RDD[Feature[Geometry, Double]] = sc.parallelize(features)
-    val keyedFeatures: RDD[(SpatialKey, Feature[Geometry, Double])] = featuresRDD.clipToGrid(target_layout)
-
-    val options = Rasterizer.Options(includePartial = true, sampleType = PixelIsPoint)
+  def vectorToRasterSpatialGeneric(path: String, targetLayout: LayoutDefinition, targetCRS: CRS, spatialBounds: KeyBounds[SpatialKey]): MultibandTileLayerRDD[SpatialKey] = {
     val cellType = DoubleConstantNoDataCellType
-    // TODO: For very large extents fromKeyedFeature gives a OOM error on the executors.
-    // TODO: This is likely because fromKeyedFeature creates an EmptyTile with (layout.tileCols, layout.tileRows) dimensions for each feature.
-    val band: RDD[(SpatialKey, Tile)] with Metadata[LayoutDefinition] = fromKeyedFeature[Geometry](keyedFeatures, cellType, target_layout, options)
-    val datacube: RDD[(SpatialKey, MultibandTile)] = band.mapValues(tile => MultibandTile(tile))
+
+    val featureBands: Seq[(String, mutable.Buffer[Feature[Geometry, Double]])] = extractFeatures(path, targetCRS, targetLayout)("")
+    val bands: RDD[(SpatialKey, MultibandTile)] = featureBands.map { case (_bandName, features) =>
+      featuresToBand(features, targetLayout, cellType)
+    }.reduce{_ union _}.groupByKey().mapValues(tiles => MultibandTile(tiles))
 
     val metadata: TileLayerMetadata[SpatialKey] = TileLayerMetadata(
       cellType,
-      target_layout,
-      target_layout.extent,
-      target_crs,
-      targetDatacube.metadata.bounds.get.toSpatial
+      targetLayout,
+      targetLayout.extent,
+      targetCRS,
+      spatialBounds
+    )
+
+    MultibandTileLayerRDD(bands, metadata)
+  }
+
+  /**
+   * Convert a vector datacube to a raster datacube with a time dimension.
+   *
+   * @param path: Path to the geojson file.
+   * @param t
+   * @return A raster datacube.
+   */
+  def vectorToRasterTemporalGeneric(path: String, targetLayout: LayoutDefinition, targetCRS: CRS, spatialBounds: KeyBounds[SpatialKey]): MultibandTileLayerRDD[SpaceTimeKey] = {
```

Comment:
the 'generic' in the name probably was meant to say that it works for bot spatial and spacetime.
Given that this does not seem to be possible, I would drop the 'generic'.